radlab-ui/webapp/src/utils/terraform.ts

```diff
@@ -217,3 +264,78 @@ const checkModuleVariablesZero = async (moduleName: string) => {
     })
   return returnModuleVariableData
 }
+
+export const checkDependsOnValid = (
+  dependsOnVarData: string | null,
+  userAnswerData: FormikValues,
+) => {
+  if (dependsOnVarData) {
+    const dependsOnDataOperatorFormats = dependsOnVarData
+      .replaceAll("&&", " && ")
+      .replaceAll("||", " || ")
+
+    const dependsOnDataAnswerMatchRes = dependsOnDataOperatorFormats
+      .split(" ")
+      .map((dependsOnDataOperatorFormat) => {
+        const checkDependsNameVar = dependsOnDataOperatorFormat.split("==")
+        let getwithAnswerMatch
+        if (checkDependsNameVar.length === 2) {
+          //@ts-ignore
+          getwithAnswerMatch = `${userAnswerData[checkDependsNameVar[0]]} == ${
+            checkDependsNameVar[1]
+          }`
+        } else {
+          getwithAnswerMatch = checkDependsNameVar[0]
+        }
+
+        return getwithAnswerMatch
+      })
+      .join(" ")
+
+    const formatDependsOnDataAnswerMatchRes = `(${dependsOnDataAnswerMatchRes.replaceAll(
+      " && ",
+      ") && (",
+    )})`
+
+    const dependsOnDataAnswerMatchEvalute = eval(
+      formatDependsOnDataAnswerMatchRes,
+    )
+
+    return dependsOnDataAnswerMatchEvalute
+  } else {
+    return false
+  }
+}
+
+export const formatRelevantVariables = (
+  formVariablesData: IUIVariable[],
+  currentAnswerValueData: FormikValues,
+) => {
+  const allNonDependsVars = formVariablesData.filter(
+    (formVariableData) => !formVariableData.dependsOn,
+  )
+  const allDependsVars = formVariablesData.filter(
+    (formVariableData) => formVariableData.dependsOn,
+  )
+
+  const findDependentVars = allDependsVars.filter((dependsVars) => {
+    const isDependsOnValid = checkDependsOnValid(
+      dependsVars.dependsOn,
+      currentAnswerValueData,
+    )
+
+    if (isDependsOnValid) {
+      return dependsVars
+    } else {
+      return null
```

Comment:
This is we are doing filter of variables data based on `checkDependsOnValid` function for user display I do not think so here required any `tests` because already we are calling here the depends check function for validation that we have the tests